compiler/modules/nback/build.bal

```diff
@@ -1106,48 +1163,67 @@ final readonly & map<bir:OrderOp> flippedOrderOps = {
 };
 
 function buildCompareTagged(llvm:Builder builder, Scaffold scaffold, bir:CompareInsn insn, llvm:Value lhs, llvm:Value rhs, bir:Register result) {
-    llvm:Value lhsIsNil = builder.iCmp("eq", lhs, llvm:constNull(llvm:pointerType("i8", 1)));
-    llvm:Value rhsIsNil = builder.iCmp("eq", rhs, llvm:constNull(llvm:pointerType("i8", 1)));
-    llvm:Value eitherNil = builder.iBitwise("or", lhsIsNil, rhsIsNil);
-
-    llvm:BasicBlock eitherNilBB = scaffold.addBasicBlock();
-    llvm:BasicBlock neitherNilBB = scaffold.addBasicBlock();
-    llvm:BasicBlock joinBB = scaffold.addBasicBlock();
-    builder.condBr(eitherNil, eitherNilBB, neitherNilBB);
-
-    builder.positionAtEnd(eitherNilBB);
-    llvm:Value bothNil = builder.iBitwise("and", lhsIsNil, rhsIsNil);
+    bir:OrderType orderTy = insn.orderType;
+    llvm:Value? compareResult = ();
+    if orderTy is bir:OptOrderType {
+        match orderTy.opt {
+            t:UT_INT => {
```

Comment:
Write a function that maps UT_INT to intCompareFunction, UT_FLOAT to floatCompareFunction. Then write this statement `compareResult = builder.call(buildRuntimeFunctionDecl(scaffold, intCompareFunction), [lhs, rhs]` once using that function.

Reply:
fixed